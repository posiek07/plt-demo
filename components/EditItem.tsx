import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { ProductDataType } from '../types';

const EditItem = ({
    numberOfItems,
    product,
    addToBasket,
    deleteFromBasket,
}: {
    numberOfItems: number;
    product: ProductDataType;
    addToBasket: (product: ProductDataType) => void;
    deleteFromBasket: (productId: number) => void;
}) => {
    return (
        <View style={styles.container}>
            <Text style={styles.bigFont} onPress={() => addToBasket(product)}>
              +
            </Text>
            <Text style={styles.bigFont}>{numberOfItems}</Text>
            <Text style={styles.bigFont} onPress={() => deleteFromBasket(product.id)}>
                -
            </Text>
        </View>
    );
};

export default EditItem;

const styles = StyleSheet.create({
    container: {
        display: 'flex',
        flexDirection: 'column',
        padding: 10,
        alignItems: 'center'
    },
    bigFont: {
        fontSize: 30,
    },
});
