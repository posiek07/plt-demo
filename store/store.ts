import { Store } from 'pullstate';

export const MyBasket = new Store({
  basket: [],
  sum: 0,
});

export const deleteFromBasket = (productId) => {
  MyBasket.update((s) => {
    const newBasket = s;
    const removedItemIndex = newBasket.basket.findIndex((item) => item.id === productId);
    newBasket.basket.splice(removedItemIndex, 1);
    newBasket.sum =
      newBasket.basket.length > 0 ? newBasket.basket.reduce((sum, { price }) => sum + price, 0) : 0;
    return newBasket;
  });
};

export const addToBasket = (product) => {
  MyBasket.update((s) => {
    const newBasket = s;
    newBasket.basket.push(product);
    newBasket.sum =
      newBasket.basket.length > 0 ? newBasket.basket.reduce((sum, { price }) => sum + price, 0) : 0;
    return newBasket;
  });
};
