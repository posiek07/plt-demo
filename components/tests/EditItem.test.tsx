import React from 'react';
import renderer from 'react-test-renderer';
import EditItem from '../EditItem';
import { addToBasket, deleteFromBasket } from '../../store/store';


const numberOfItems = 5;

test('renders correctly', () => {
    const tree = renderer
        .create(
            <EditItem
                numberOfItems={numberOfItems}
                product={{ id: 3, colour: 'Blue', img: 'link', name: 'blue dress', price: 3 }}
                addToBasket={addToBasket}
                deleteFromBasket={deleteFromBasket}

            />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
