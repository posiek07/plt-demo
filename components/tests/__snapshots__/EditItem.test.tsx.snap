// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "padding": 10,
    }
  }
>
  <Text
    onPress={[Function]}
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    +
  </Text>
  <Text
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    5
  </Text>
  <Text
    onPress={[Function]}
    style={
      Object {
        "fontSize": 30,
      }
    }
  >
    -
  </Text>
</View>
`;
